core/blockchain.go

```diff
@@ -365,14 +365,55 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, genesis *Genesis
 		diffQueue:          prque.New[int64, *types.DiffLayer](nil),
 		diffQueueBuffer:    make(chan *types.DiffLayer),
 	}
+	var err error
+
+	// Open trie database with provided config
+	var triedb *trie.Database
+	if cacheConfig.SeparateTrieConfig != nil {
+		separatedTrieConfig := cacheConfig.SeparateTrieConfig
+		separatedTrieDir := separatedTrieConfig.TrieDataDir
+		log.Info("node run with separated trie database", "directory", separatedTrieDir)
+		// open the separated db to init the trie database which only store the trie data
+		separateDB, dbErr := rawdb.Open(rawdb.OpenOptions{
+			Type:              separatedTrieConfig.SeparateDBEngine,
+			Directory:         separatedTrieDir,
+			AncientsDirectory: filepath.Join(separatedTrieDir, separatedTrieConfig.SeparateDBAncient),
```

Comment:
Are these parameters for separatedDB really necessary?  

- `Directory`
- `AncientsDirectory`
- `Namespace`
- `Cache`
- `Handle`


I think most of those can be default or calculated from the original parameters(`triedbConfig`)

Reply:
no,  it calculated by the config of node stack  which can not obtain in this function .   triedbConfig does not contain this information